src/test/test_utils.rs

```diff
@@ -8,22 +8,22 @@
 use std::fmt;
 
 #[derive(Debug, Copy, Clone, PartialEq)]
-pub enum ShadowPassing {
-    Yes,
-    No,
+pub enum TestEnvironment {
+    Shadow,
+    Libc,
 }
 
 pub struct ShadowTest<T, E> {
     name: String,
     func: Box<dyn Fn() -> Result<T, E>>,
-    shadow_passing: ShadowPassing,
+    passing: Vec<TestEnvironment>,
```

Comment:
Optional - it's a *little* weird to use a `Vec` when conceptually it's representing a set. Is there a reason not to use `HashSet`?

Reply:
I used `Vec` to reduce the verbosity since it makes the test cases easier to read (`vec![A, B]` vs `[A, B].iter().cloned().collect()` for each test case), but I've changed it to a `HashSet` since it is the right data type.

Reply:
Looking at it again, just using two separate fields might make the most sense since we don't expect the number of possible environments to grow beyond 2.

I'm fine with leaving as you have it now, and agree the extra verbosity is annoying but not enough to prefer a `Vec`. I wouldn't be opposed to writing a `set!` macro analogous to the `vec!` macro :)

Reply:
I'm a little worried that in the future we might also want "Shadow-ptrace" and "Shadow-preload" options as well if a test only works under one of the runtimes. So I think the flexibility of using a set is good for now, but we can revisit it later.